tools/precommit.py

```diff
@@ -0,0 +1,141 @@
+        fs.chdir('apps')
+        ex.check_run_cmd('git', ['pull'])
+        fs.chdir('..')
+    else:
+        ex.check_run_cmd('git', ['clone',
+                                 'https://bitbucket.org/nuttx/apps.git'])
+
+    # Step 2
+    fs.maybe_make_directory(fs.join(nuttx_root, 'apps', 'system', 'iotjs'))
+    for file in fs.listdir(fs.join(path.PROJECT_ROOT,
+                                   'targets','nuttx-stm32f4','app')):
+        fs.copy(fs.join(path.PROJECT_ROOT,'targets','nuttx-stm32f4','app',file),
+                fs.join(nuttx_root, 'apps', 'system', 'iotjs'))
+
+    # Step 3
+    fs.chdir(fs.join(nuttx_root, 'nuttx', 'tools'))
+    ex.check_run_cmd('./configure.sh', ['stm32f4discovery/usbnsh'])
+    fs.chdir('..')
+    fs.copy(fs.join(path.PROJECT_ROOT,
+                    'targets',
+                    'nuttx-stm32f4',
+                    'nuttx',
+                    '.config.default'),
+            fs.join('.config'))
+
+
+def build_nuttx(nuttx_root, buildtype):
+    fs.chdir(fs.join(nuttx_root, 'nuttx'))
+    try:
+        nuttx_bin = fs.join(nuttx_root, 'nuttx', 'nuttx.bin')
+        if fs.exists(nuttx_bin):
+            fs.remove(nuttx_bin)
+        if buildtype == "release":
+            ex.run_cmd('make', ['IOTJS_ROOT_DIR=' + path.PROJECT_ROOT, 'R=1'])
+        else:
+            ex.run_cmd('make', ['IOTJS_ROOT_DIR=' + path.PROJECT_ROOT, 'R=0'])
+
+        if fs.exists(nuttx_bin):
+            return True
+        else:
+            print "Cannot create nuttx.bin"
+            return False
+    except OSError as err:
+        print err
+        return False
+
+
+def build(buildtype, args):
+    fs.chdir(path.PROJECT_ROOT)
+    ex.check_run_cmd('./tools/build.py', args + ['--buildtype=' + buildtype])
+
+
+option = parse_option()
+if os.getenv('TRAVIS') == None:
+    ex.check_run_cmd('tools/check_signed_off.sh')
+
+for test in option.test:
+    if test == "host":
+        for buildtype in option.buildtype:
+            build(buildtype, [])
+
+    elif test == "rpi2":
+        for buildtype in option.buildtype:
+            build(buildtype, ['--target-arch=arm',
+                              '--target-board=rpi2'])
+
+    elif test == "nuttx":
+        for buildtype in option.buildtype:
+            nuttx_root=fs.join(path.PROJECT_ROOT, 'deps', 'nuttx')
+            setup_nuttx_root(nuttx_root)
+            build_nuttx(nuttx_root, buildtype)
+            build(buildtype, ['--target-arch=arm',
+                              '--target-os=nuttx',
+                              '--nuttx-home=' + fs.join(nuttx_root, 'nuttx'),
+                              '--target-board=stm32f4dis',
+                              '--jerry-heaplimit=78'])
+            if not build_nuttx(nuttx_root, buildtype):
+                ex.fail('nuttx ' + buildtype + ' build failed')
+
+    elif test == "misc":
+        build("debug", ['--no-snapshot', '--jerry-lto'])
+        build("debug", ['--iotjs-minimal-profile'])
+
+    else:
```

Comment:
I think we need a more delicate error message for `darwin` as long as we specify to support platform like above. ['host', 'rpi2', 'nuttx', 'darwin', 'misc']